### ISSUE TITLE
feature:  Allow to use HTTP other Method (GET, PUT, DELETE) to impl kserve predict protocol v2

### ISSUE THREAD

**Issue by KimSoungRyoul** [bentoml/BentoML#4590]
### Feature request

https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md

~~~python

svc = bentoml.Service(...)

@svc.api(
    route="predict/${PATH_VARIABLE}",
    method="GET", # <----
    
)
async def predict(path_variable: str):
     ....

~~~

**Comment by KimSoungRyoul** [comment]
hi @frostming 

I think that support other Method (as least `GET`) not only `POST` in BentoML is useful to extend BentoML

Can you include this issue in bentoml team's maelstrom or 
if you're positive about this minor feature, I can make the PR

thanks

**Comment by alvaro-stylesage** [comment]
Hello, are there any updates on this issue? Is it currently possible to define an endpoint with `@bentoml.api` that will be consumed via GET method? Thanks a lot for the update.

**Comment by frostming** [comment]
> Hello, are there any updates on this issue? Is it currently possible to define an endpoint with `@bentoml.api` that will be consumed via GET method? Thanks a lot for the update.

No, but you are able to mount a FastAPI app providing GET endpoints

https://docs.bentoml.com/en/latest/build-with-bentoml/asgi.html

**Comment by aarnphm** [comment]
I think we will go ahead and refer implementation in FastAPI here.
